app/services/datastore/get_application.rb

```diff
@@ -7,14 +7,15 @@ def initialize(application_id)
     end
 
     def call
-      app = DatastoreApi::Requests::GetApplication.new(
-        application_id:
-      ).call
+      app = Rails.error.record do
+        # Will report and re-raise the exception
+        DatastoreApi::Requests::GetApplication.new(application_id:).call
+      end
 
       raise DatastoreApi::Errors::NotFoundError if superseded?(app)
 
       app
-    rescue DatastoreApi::Errors::NotFoundError
+    rescue DatastoreApi::Errors::ApiError
```

Comment:
Did you intent to change the error class here?

Reply:
Yep, `NotFoundError` is a subclass of `ApiError`, but there are other errors that can be raised on the Datastore that subclass from `ApiError` that had the risk of not being caught if using a more specific class (like `NotFoundError` before), so I decided it is best to use a more generic class so we are better covered.

Reply:
👍